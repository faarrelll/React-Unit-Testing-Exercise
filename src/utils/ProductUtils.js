export const product = {
    id: 1,
    name: "Nasi Goreng Cak Kodir Special Telur Biawak",
    price: 1500000000,
    category : "Makanan",
    image: "nasgor.jpg",
  };