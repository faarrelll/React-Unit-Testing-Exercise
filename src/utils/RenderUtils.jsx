import { render } from "@testing-library/react";
import Products from "../pages/Products";
import ProductModal from "../components/ProductModal";
import { LanguageProvider } from "../context/LanguageContext";
import { BrowserRouter } from "react-router";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import { AuthProvider } from "../context/AuthContext";

const renderProducts = () => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <Provider store={store}>
          <LanguageProvider>
            <Products />
          </LanguageProvider>
        </Provider>
      </AuthProvider>
    </BrowserRouter>
  );
};

const renderProductsModal = ({
  open,
  onClose,
  product,
  quantity,
  onQuantityIncrease,
  onQuantityDecrease,
}) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <Provider store={store}>
          <LanguageProvider>
            <ProductModal
              open={open}
              onClose={onClose}
              product={product}
              quantity={quantity}
              onQuantityIncrease={onQuantityIncrease}
              onQuantityDecrease={onQuantityDecrease}
            />
          </LanguageProvider>
        </Provider>
      </AuthProvider>
    </BrowserRouter>
  );
};

export { renderProducts, renderProductsModal };
