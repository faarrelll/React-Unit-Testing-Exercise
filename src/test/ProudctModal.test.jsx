import { describe, it, expect } from "vitest";
import { renderProductsModal } from "../utils/RenderUtils";
import { product } from "../utils/ProductUtils";

describe("ProductModal Component", () => {
  it("should match the snapshot", async () => {
    const container = renderProductsModal({
      open: true,
      onClose: () => {},
      product: product,
      quantity: 1,
      onQuantityIncrease: () => {},
      onQuantityDecrease: () => {},
    });
    expect(container).toMatchSnapshot();
  });
});
