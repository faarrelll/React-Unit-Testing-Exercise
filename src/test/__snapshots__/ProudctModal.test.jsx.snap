// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductModal Component > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      aria-describedby="modal-description"
      aria-labelledby="modal-title"
      class="MuiModal-root css-1g04pbp-MuiModal-root"
      role="dialog"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-4nmryk-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="modal-content"
        tabindex="-1"
      >
        <div
          class="image-section"
        >
          <img
            alt="Nasi Goreng Cak Kodir Special Telur Biawak"
            class="product-image"
            src="nasgor.jpg"
          />
        </div>
        <div
          class="content-section"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 product-title css-1miy0lu-MuiTypography-root"
          >
            Nasi Goreng Cak Kodir Special Telur Biawak
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 product-category css-7ctihp-MuiTypography-root"
          >
            Makanan
          </p>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-1jpc804-MuiDivider-root"
          />
          <h6
            class="MuiTypography-root MuiTypography-h6 product-price css-1miy0lu-MuiTypography-root"
          >
            Rp 
            1,500,000,000
          </h6>
          <div
            class="quantity-control"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall quantity-button css-r363y4-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="RemoveIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13H5v-2h14z"
                />
              </svg>
            </button>
            <p
              class="MuiTypography-root MuiTypography-body1 quantity-text css-rizt0-MuiTypography-root"
            >
              1
            </p>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall quantity-button css-r363y4-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
                />
              </svg>
            </button>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 total-price css-rizt0-MuiTypography-root"
          >
            Total: Rp 
            1,500,000,000
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary add-to-cart-button css-1abvrni-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Tambah ke Keranjang
          </button>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
